ambry-router/src/main/java/com.github.ambry.router/DefaultKeyManagementServiceFactory.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.router;
+
+import com.github.ambry.config.ClusterMapConfig;
+import com.github.ambry.config.KMSConfig;
+import com.github.ambry.config.VerifiableProperties;
+import javax.crypto.spec.SecretKeySpec;
+
+
+/**
+ * {@link DefaultKeyManagementServiceFactory} specific implementation of {@link KeyManagementServiceFactory}.
+ * <p/>
+ * Sets up all the supporting cast required for the operation of {@link DefaultKeyManagementService} and returns a new instance on
+ * {@link #getKeyManagementService()}.
+ */
+public class DefaultKeyManagementServiceFactory implements KeyManagementServiceFactory<SecretKeySpec> {
+  private final KMSConfig kmsConfig;
+  private final ClusterMapConfig clusterMapConfig;
+
+  DefaultKeyManagementServiceFactory(VerifiableProperties verifiableProperties) {
+    kmsConfig = new KMSConfig(verifiableProperties);
+    clusterMapConfig = new ClusterMapConfig(verifiableProperties);
```

Comment:
if cluster name is all that is required, then that is what should be stored/passed to the constructor of `DefaultKeyManagementService`.

Related question, can the factory take `KMSConfig` and cluster name instead of `VerifiableProperties`? (like `ClusterAgentsFactory` for e.g.).